src/backends/plonky2/primitives/merkletree_circuit.rs

```diff
@@ -154,6 +154,8 @@ impl MerkleProofGadget {
 
 impl MerkleClaimAndProofTarget {
     /// assigns the given values to the targets
+    // @arnaucube, can we define a MerkleClaimAndProofInput for this?
```

Comment:
@arnaucube this comment is for you

Reply:
I'm in favor of not bloating the code with a custom struct type for each method that takes multiple inputs, bcs it's extra overhead to deal with, and also would mean having to account for types that we actually don't need. Therefore I'm in favor of using the `#[allow(clippy::too_many_arguments)]` to avoid being forced to bloat the code.